### PR TITLE
Fix some threading issues with tcc backend

### PR DESCRIPTION
**Pull request by vegansk**
The code compiles again with tcc backend.

### CODE REVIEW

File: lib/nimbase.h [hunk 1]
```diff
@@ -108,6 +108,8 @@ __clang__
        defined __SUNPRO_C || \
        defined __xlC__
 #  define NIM_THREADVAR __thread
+#elif defined __TINYC__
+#  defined NIM_THREADVAR
```
Comment (yglukhov) [inline lib/nimbase.h]: Is this a typo? Should be `define` instead of `defined`?

Comment (vegansk) [inline lib/nimbase.h]: Oops, of cause ``#  define NIM_THREADVAR`` :-(  Can you commit the fix directly in nim-lang repo, or should I make a PR?

Comment (yglukhov) [inline lib/nimbase.h]: Dont have commit access. Please do a pr.

Comment (def-) [inline lib/nimbase.h]: I commited the fix, thanks for reporting.

**Pull request merged**